cmd/genyaml/gen_kubectl_yaml.go

```diff
@@ -29,6 +30,15 @@ import (
 	"k8s.io/kubernetes/pkg/kubectl/cmd"
 )
 
+type devNull int
+
+// nullReader is something like iotil.Discard, but a Reader
+var nullReader io.Reader = devNull(0)
+
+func (devNull) Read(p []byte) (int, error) {
+	return len(p), nil
```

Comment:
You have written code that accurately implements the old comment... however, the old comment did not describe how os.Stdin is likely to work. If nothing was passed to stdin, I expect it would have returned 0 bytes and an EOF error when called. I have a hard time imagining an infinite stream of zeros will be handled correctly by the rest of the code :) Is there no test that is affected by this?

Reply:
The previous changes seem does not affect the tests :), I updated the commits and moved the `devNull` and related code into `genutils.go` file and add a few more tests to verify it. PTAL 😄